Rakefile

```diff
@@ -29,3 +36,87 @@ RSpec::Core::RakeTask.new(:spec) do |r|
 end
 
 task :default => :spec
+
+
+class EmptyMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    if @app.respond_to?(:call)
+      status, headers, body = @app.call(env)
+    else
+      [200, {}, 'Hello world']
+    end
+  end
+end
+
+class FaradayMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    conn = Faraday.new(url: 'http://www.google.com') do |faraday|
```

Comment:
you're actually benchmarking against google.com? what about webmock or something? what if google is slow in your geographical region or something?